cpp/include/raft/random/rng.hpp

```diff
@@ -18,6 +18,20 @@
  * Please use the cuh version instead.
  */
 
+/**
+ * DISCLAIMER: this file is deprecated: use rng.cuh instead
+ */
+
 #pragma once
 
-#include "rng.cuh"
\ No newline at end of file
+#if defined(_MSC_VER)
+#pragma message(                                                               \
+  "raft/random/rng.hpp is deprecated and will be removed in a future relase. " \
+  "Please use raft/random/rng_state.hpp or raft/random/rng.cuh instead")
+#else
+#warning \
```

Comment:
I had held off on doing a warning only because I Know this is immediately going to break a few consumers downstream who have enabled "warnings as errors". For this reason, I'm okay removing the warning tag.

Reply:
Indeed the builds weren't working, so I removed this warning again

Reply:
Actually, I'm using the same mechanism as in `raft/spatial/knn.hpp` now